monitor.cpp

```diff
@@ -18,37 +55,70 @@ void displayWarning() {
 /// @param pulseRate
 /// @param spo2
 /// @return 1 if vitals are ok, 0 otherwise
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (checkVitals(temperature, pulseRate, spo2)) {
-    return 1;
-  }
-  return 0;
+bool isVitalsOk(float temperature, float pulseRate, float spo2) {
+  initializeWarnigMessages();
+  return (isTemperatureOk(temperature) && isPulseRateOk(pulseRate) && isSpo2Ok(spo2));
+}
+
+bool isItInRange(float value, float lower, float upper) {
+     return (value >= lower && value <= upper);
 }
 
-bool checkVitals(float temperature, float pulseRate, float spo2) {
-  return (checkTemperature(temperature) && checkPulseRate(pulseRate) && checkSpo2(spo2));
+void displayMessage(int MsgInd, Language lang) {
+  // Set console output to UTF-8
+  // SetConsoleOutputCP(CP_UTF8);
+    system("chcp 65001");
+    if (MsgInd > 4) {
+        MsgInd = 4;
+    }
+    // Print the wide string
+    std::wcout.imbue(std::locale(std::locale(), new std::codecvt_utf8<wchar_t>));
+    std::wcout << warningMessages[lang][MsgInd] << std::endl;
+}
+void displayTemperatureWarning(float temperature, Language lang) {
+     int MsgInd = 0;
+     std::map<float, float> temperatureRange = { {0, 95}, {95, 96.53},
+                                                {96.54, 100.47}, {100.48, 102}, {102, 102} };
+      for (auto itr = temperatureRange.begin(); itr != temperatureRange.end(); ++itr) {
+          if (isItInRange(temperature, itr->first, itr->second)) {
+              break;
+          }
+          MsgInd++;
```

Comment:
The `MsgInd` is an index that has to be the same between `temperatureRange` and the array of messages inside `warningMessages`.
Whenever this happens, it may be an opportunity to put things together. How about a `struct` that puts the ranges together with the classification (`TOO_LOW`, `WARNING_LOW`, etc) and this function makes the classification. Then the classification is converted into a message using a map.
How does that sound?

Reply:
used the map of map for storing the warning messages, and map of vector for ranges of temperature. Thanks for commenting on it.